pkg/ccloudv2/ccpm.go

```diff
@@ -0,0 +1,120 @@
+		if err != nil {
+			return nil, errors.CatchCCloudV2Error(err, httpResp)
+		}
+		allPlugins = append(allPlugins, page.GetData()...)
+		pageToken, done, err = extractNextPageToken(page.GetMetadata().Next)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return allPlugins, nil
+}
+
+func (c *Client) DescribeCCPMPlugin(id, environment string) (ccpmv1.CcpmV1CustomConnectPlugin, error) {
+	resp, httpResp, err := c.CCPMClient.CustomConnectPluginsCcpmV1Api.GetCcpmV1CustomConnectPlugin(c.ccpmApiContext(), id).Environment(environment).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) DeleteCCPMPlugin(id, environment string) error {
+	httpResp, err := c.CCPMClient.CustomConnectPluginsCcpmV1Api.DeleteCcpmV1CustomConnectPlugin(c.ccpmApiContext(), id).Environment(environment).Execute()
+	return errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) UpdateCCPMPlugin(id string, request ccpmv1.CcpmV1CustomConnectPluginUpdate) (ccpmv1.CcpmV1CustomConnectPlugin, error) {
+	resp, httpResp, err := c.CCPMClient.CustomConnectPluginsCcpmV1Api.UpdateCcpmV1CustomConnectPlugin(c.ccpmApiContext(), id).CcpmV1CustomConnectPluginUpdate(request).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) executeListCCPMPlugins(pageToken, cloud, environment string) (ccpmv1.CcpmV1CustomConnectPluginList, *http.Response, error) {
+	req := c.CCPMClient.CustomConnectPluginsCcpmV1Api.ListCcpmV1CustomConnectPlugins(c.ccpmApiContext()).PageSize(ccloudV2ListPageSize)
+	if pageToken != "" {
+		req = req.PageToken(pageToken)
+	}
+	if cloud != "" {
+		req = req.SpecCloud(cloud)
+	}
+	if environment != "" {
+		req = req.Environment(environment)
+	}
+	return req.Execute()
+}
+
+func (c *Client) CreateCCPMPresignedUrl(request ccpmv1.CcpmV1PresignedUrl) (ccpmv1.CcpmV1PresignedUrl, error) {
+	resp, httpResp, err := c.CCPMClient.PresignedUrlsCcpmV1Api.CreateCcpmV1PresignedUrl(c.ccpmApiContext()).CcpmV1PresignedUrl(request).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) CreateCCPMPluginVersion(pluginId string, request ccpmv1.CcpmV1CustomConnectPluginVersion) (ccpmv1.CcpmV1CustomConnectPluginVersion, error) {
+	resp, httpResp, err := c.CCPMClient.CustomConnectPluginVersionsCcpmV1Api.CreateCcpmV1CustomConnectPluginVersion(c.ccpmApiContext(), pluginId).CcpmV1CustomConnectPluginVersion(request).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) DescribeCCPMPluginVersion(pluginId, versionId, environment string) (ccpmv1.CcpmV1CustomConnectPluginVersion, error) {
+	resp, httpResp, err := c.CCPMClient.CustomConnectPluginVersionsCcpmV1Api.GetCcpmV1CustomConnectPluginVersion(c.ccpmApiContext(), pluginId, versionId).Environment(environment).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) ListCCPMPluginVersions(pluginId, environment string) ([]ccpmv1.CcpmV1CustomConnectPluginVersion, error) {
```

Comment:
Since it looks like plugin versions don't support pagination, we can write this like the functions, without a loop.

And to double check, is it intentional that this API doesn't support pagination?